src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/annotation/annotationonsameline/InputAnnotationOnSameLinePatternVariables.java

```diff
@@ -0,0 +1,40 @@
+/*
+AnnotationOnSameLine
+tokens = (default)CLASS_DEF, INTERFACE_DEF, ENUM_DEF, METHOD_DEF, CTOR_DEF, \
+         VARIABLE_DEF, RECORD_DEF, COMPACT_CTOR_DEF
+
+
+*/
+
+//non-compiled with javac: Compilable with Java21
+package com.puppycrawl.tools.checkstyle.checks.annotation.annotationonsameline;
+
+public class InputAnnotationOnSameLinePatternVariables {
```

Comment:
Let's add some unnamed variables in here too: `@Deprecated int _ = 0;` as well as unnamed record components.

Reply:
done